nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/flak.dm

```diff
@@ -0,0 +1,228 @@
+		last_target = null
+	else
+		fire_weapon(last_target, mode=FIRE_MODE_FLAK, lateral=TRUE)
+		return
+	for(var/obj/structure/overmap/ship in GLOB.overmap_objects)
+		if(!ship || !istype(ship))
+			continue
+		if(ship == src || ship.faction == faction || ship.wrecked || ship.z != z) //No friendly fire, don't blow up wrecks that the crew may wish to loot.
+			continue
+		var/target_range = get_dist(ship,src)
+		if(target_range > initial(weapon_range)+30)
+			continue
+		if(!QDELETED(ship) && isovermap(ship))
+			if(mass >= MASS_MEDIUM)
+				fire_weapon(ship, mode=FIRE_MODE_FLAK, lateral=TRUE)
+				break
+			else
+				fire_weapon(ship, mode=FIRE_MODE_PDC, lateral=TRUE)
+				break
+
+/obj/structure/overmap/proc/get_flak_range(atom/target)
+	if(!target)
+		target = src
+	var/dist = (get_dist(src, target) / 2)
+	var/minimum_safe_distance = pixel_collision_size_y / 32
+	return (dist >= minimum_safe_distance) ? dist : minimum_safe_distance //Stops you flak-ing yourself
+
+/obj/structure/overmap/proc/fire_flak(target,speed=null)
+	var/turf/T = get_turf(src)
+	var/flak_range = get_flak_range(target)
+	var/obj/item/projectile/proj = new /obj/item/projectile/bullet/flak(T, flak_range)
+	proj.starting = T
+	if(gunner)
+		proj.firer = gunner
+	else
+		proj.firer = src
+	proj.def_zone = "chest"
+	proj.original = target
+	proj.pixel_x = round(pixel_x)
+	proj.pixel_y = round(pixel_y)
+	var/theangle = Get_Angle(src,target)
+	spawn()
+		proj.fire(theangle)
+		if(speed)
+			proj.set_pixel_speed(speed)
+
+/obj/structure/overmap/proc/handle_flak()
+	if(mass <= MASS_MEDIUM) //This is for big boys only.
+		return
+
+/obj/effect/temp_visual/flak
+	icon = 'nsv13/goonstation/icons/effects/explosions/80x80.dmi'
+	icon_state = "explosion"
+	duration = 2 SECONDS
+	pixel_x = -32
+	pixel_y = -32
+	var/flak_range = 2 //AOE where flak hits torpedoes. May need to buff this a bit.
+
+/obj/item/projectile/bullet/flak
+	icon_state = "flak"
+	name = "flak round"
+	damage = 2
+	alpha = 0
+	var/steps_left = 0 //Flak range, AKA how many tiles can we move before we go kaboom
+
+/obj/item/projectile/bullet/flak/Initialize(mapload, range=10)
+	. = ..()
+	steps_left = range
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, .proc/check_range)
+
+/obj/item/projectile/bullet/flak/proc/explode()
+	new /obj/effect/flak_handler(get_turf(src))
+	qdel(src)
+
+//Small object to make flak "flicker" a bit. Kills itself after spawning flak
+/obj/effect/flak_handler/Initialize()
+	. = ..()
+	for(var/I = 0, I < rand(2,5), I++)
+		var/edir = pick(GLOB.alldirs)
+		new /obj/effect/temp_visual/flak(get_turf(get_step(src, edir)))
+		sleep(rand(0, 2))
+	return INITIALIZE_HINT_QDEL
+
+/obj/effect/temp_visual/flak/Initialize()
+	if(prob(50))
+		icon = 'nsv13/goonstation/icons/effects/explosions/96x96.dmi'
+	for(var/obj/X in view(flak_range, src))
+		var/severity = flak_range-get_dist(X, src)
+		if(istype(X, /obj/structure))
+			X.take_damage(severity*10, damage_type = "overmap_light")
```

Comment:
Overmap_light is not a damage type.
It is an armour.

Reply:
Wack.

Well it ought to be..but that explains a lot, lol

Reply:
300% sure I've told you this before.